test/unit/src/index-config-test.js

```diff
@@ -205,3 +205,65 @@ test('Configuration - validation', async t => {
     reset()
   }
 })
+
+test('Configuration - service verification', async t => {
+  t.plan(8)
+  let aws
+  let started = await server.start()
+  t.ok(started, 'Started server')
+
+  // Default behavior: verification enabled
+  aws = await client(config)
+  try {
+    await aws({ service: 'not-a-service' })
+    t.fail('Should throw on unverified service name')
+  }
+  catch (err) {
+    t.match(err.message, /Invalid AWS service/, 'By default, throw on unverified service name')
+  }
+  // Disable service verification at request time
+  try {
+    await aws({ verifyService: false, service: 'not-a-service' })
+    t.pass('Service verification disabled')
+  }
+  catch (err) {
+    t.fail(err)
+  }
+  // Service name still required
+  try {
+    await aws({ verifyService: false })
+    t.fail('Should throw on missing service name')
+  }
+  catch (err) {
+    t.match(err.message, /No AWS service specified/, 'Throw on missing service name')
+  }
+
+  // Disable verification at client
+  aws = await client({ ...config, verifyService: false })
+  try {
+    await aws({ service: 'not-a-service' })
+    t.pass('Service verification disabled')
+  }
+  catch (err) {
+    t.fail(err)
+  }
+  // Request can enable service verification
+  try {
+    await aws({ verifyService: true, service: 'not-a-service' })
+    t.fail('Should throw on unverified service name')
+  }
+  catch (err) {
+    t.match(err.message, /Invalid AWS service/, 'Throw on unverified service name')
+  }
+  // Service name still required
+  try {
+    await aws({ verifyService: true })
+    t.fail('Should throw on missing service name')
+  }
+  catch (err) {
+    t.match(err.message, /No AWS service specified/, 'Throw on missing service name')
+  }
+
```

Comment:
Did I miss a case for the client defaulting to verifying service, and an individual request disables verification and supplies a non-standard service?

Reply:
Yes, the second assertion does this: https://github.com/architect/aws-lite/pull/97/files/3fa6aa6ce27f61064d74fa351f5598fb31d92621#diff-bec2ce27d1b6c5f162499b2dd54ebdd787eca37b4c4e38d4898340c7bd316944R224-R231

That client was created without the `verifyService` option, so it is defaulted to true. 
First I make sure a request with that default client throws. 
Then I make another request with `{ verifyService: false, service: 'not-a-service' }` to make sure it doesn't throw.

Is that the case you're thinking of?